task/prefetch-dependencies-oci-ta/0.1/prefetch-dependencies-oci-ta.yaml

```diff
@@ -381,6 +277,33 @@ spec:
           update-ca-trust
         fi
 
+        # RHSM HANDLING: REGISTER RHSM & CACHI2 CONFIGURATION
+        if [ -e /activation-key/org ]; then
+          RHSM_ORG=$(cat /activation-key/org)
+          RHSM_ACT_KEY=$(cat /activation-key/activationkey)
+
+          echo "Registering with Red Hat subscription manager."
+          subscription-manager register \
+            --org "${RHSM_ORG}" \
+            --activationkey "${RHSM_ACT_KEY}" || exit 1
+
```

Comment:
IMO the cleanest way to ensure unregistration would be to `trap 'subscription-manager unregister' EXIT` just below the register call

No need to check the existence of `/activation-key/org` in the trap, because we'd only be setting up the trap in cases where it's really needed. And no need to handle unregistration again at the end of the task script

Reply:
> IMO the cleanest way to ensure unregistration would be to trap 'subscription-manager unregister' EXIT just below the register call

That is cleaner, then again, unregister itself can fail which would make the whole task fail. Unregistering is a good practice, but I think it should be best effort. I mean I wouldn't want to force users to rerun pipelines just because there's some intermittent error on the unregister endpoint, so I'd like to keep it as a function. I'll drop the check as you suggest and move the trap call right after the register sequence.

Reply:
:+1: 

For what it's worth, I think that traps don't change the overall exit status unless you explicitly call `exit N` in the trap itself. But it's best to be explicit about it, one never knows with Bash. (+1 to keep the function)